auth/pat.go

```diff
@@ -742,22 +373,22 @@ type PATS interface {
 	RevokePATSecret(ctx context.Context, token, patID string) error
 
 	// AddScope function adds a new scope entry.
-	AddPATScopeEntry(ctx context.Context, token, patID string, platformEntityType PlatformEntityType, optionalDomainID string, optionalDomainEntityType DomainEntityType, operation OperationType, entityIDs ...string) (Scope, error)
+	AddScopeEntry(ctx context.Context, token, patID string, scope []Scope) (ScopesPage, error)
 
 	// RemoveScope function removes a scope entry.
-	RemovePATScopeEntry(ctx context.Context, token, patID string, platformEntityType PlatformEntityType, optionalDomainID string, optionalDomainEntityType DomainEntityType, operation OperationType, entityIDs ...string) (Scope, error)
+	RemoveScopeEntry(ctx context.Context, token, patID string, scope []Scope) (ScopesPage, error)
 
 	// ClearAllScope function removes all scope entry.
-	ClearPATAllScopeEntry(ctx context.Context, token, patID string) error
+	ClearAllScopeEntry(ctx context.Context, token, patID string) error
 
 	// IdentifyPAT function will valid the secret.
 	IdentifyPAT(ctx context.Context, paToken string) (PAT, error)
 
 	// AuthorizePAT function will valid the secret and check the given scope exists.
-	AuthorizePAT(ctx context.Context, userID, patID string, platformEntityType PlatformEntityType, optionalDomainID string, optionalDomainEntityType DomainEntityType, operation OperationType, entityIDs ...string) error
+	AuthorizePAT(ctx context.Context, userID, patID string, entityType EntityType, optionalDomainID string, operation Operation, entityIDs ...string) error
 
```

Comment:
`AuthorizePAT` should work for one scope entityID not for many entityIDs 
```suggestion
	AuthorizePAT(ctx context.Context, userID, patID string, entityType EntityType, optionalDomainID string, operation Operation, entityID string) error
```